precovery/sourcecatalog.py

```diff
@@ -129,19 +123,30 @@ def iterate_observations(filename: str) -> Iterator[SourceObservation]:
             read_observations = progress.add_task(
                 "loading observations...", total=n_rows
             )
-            for row in table.iterrows():
-                exposure_id = row["exposure"]
-                obscode = _obscode_from_exposure_id(exposure_id)
-                id = row["id"]
-                dec = row["values_block_0"][1]
-                ra = row["values_block_0"][10]
-
-                epoch = row["values_block_0"][9]
-
-                obs = SourceObservation(exposure_id, obscode, id, ra, dec, epoch)
-                yield (obs)
-                progress.update(read_observations, advance=1)
-
+            for chunk in store.select(
+                key=key,
+                iterator=True,
+                chunksize=chunksize,
+                columns=["obs_id", "exposure_id", "mjd_utc", "ra", "dec"]
+            ):
+                exposure_ids = chunk.exposure_id.values
+                ids = chunk.obs_id.values
+                decs = chunk.dec.values
+                ras = chunk.ra.values
+                epochs = chunk.mjd_utc.values
+
+                for exposure_id, id, ra, dec, epoch in zip(
+                    exposure_ids,
+                    ids,
+                    ras,
+                    decs,
+                    epochs
+                ):
+                    obscode = _obscode_from_exposure_id(exposure_id.encode())
+
+                    obs = SourceObservation(exposure_id, obscode, id.encode(), ra, dec, epoch)
```

Comment:
Probably needs to be `exposure_id.encode()` in here, since it's a string, not raw bytes.

Reply:
https://github.com/B612-Asteroid-Institute/precovery/blob/74e113b3b29f1875e06716df0821b5e4252f6d6f/precovery/sourcecatalog.py#L11-L18 

https://github.com/B612-Asteroid-Institute/precovery/blob/74e113b3b29f1875e06716df0821b5e4252f6d6f/precovery/sourcecatalog.py#L151-L167

`SourceObservation` takes an exposure_id as a string while `_obscode_from_exposure_id` takes the exposure_id as bytes, so I think what is currently implemented is correct? 

Come to think of it, we encode and then decode the exposure_id when we call `_obscode_from_exposure_id`. I can modify `_obscode_from_exposure_id` so it just takes a string.

Reply:
Ah you are right of course; I always get encode and decode mixed up.